internal/hook/wait_exec_hook_handler.go

```diff
@@ -166,7 +166,11 @@ func (e *DefaultWaitExecHookHandler) HandleHooks(
 					err := fmt.Errorf("hook %s in container %s expired before executing", hook.HookName, hook.Hook.Container)
 					hookLog.Error(err)
 					errors = append(errors, err)
-					hookTracker.Record(newPod.Namespace, newPod.Name, hook.Hook.Container, hook.HookSource, hook.HookName, hookPhase(""), true)
+
+					trackerErr := hookTracker.Record(newPod.Namespace, newPod.Name, hook.Hook.Container, hook.HookSource, hook.HookName, hookPhase(""), true)
```

Comment:
Any reason you are ignoring the errors returned by `Record` in `internal/hook/item_hook_handler.go`
but handling the errors in this file?

Reply:
The `Record` error was ignored in `internal/hook/item_hook_handler.go` just for simplicity, which I consider might not be a good behavior. 

Will change it.